ext/standard/url.c

```diff
@@ -93,15 +93,15 @@ static const char *binary_strcspn(const char *s, const char *e, const char *char
 }
 
 /* {{{ php_url_parse */
-PHPAPI php_url *php_url_parse_ex(char const *str, size_t length)
+PHPAPI php_url *php_url_parse_ex(char const *str, size_t length, bool strict)
 {
 	bool has_port;
-	return php_url_parse_ex2(str, length, &has_port);
+	return php_url_parse_ex2(str, length, strict, &has_port);
 }
 
 /* {{{ php_url_parse_ex2
  */
-PHPAPI php_url *php_url_parse_ex2(char const *str, size_t length, bool *has_port)
+PHPAPI php_url *php_url_parse_ex2(char const *str, size_t length, bool strict, bool *has_port)
```

Comment:
The naming is getting out of hand, and we might want to start removing or deprecating the version which just passes a ``char *`` (which now that I stare at I don't know if it's the content of the str that is constant or the pointer address that is constant...). Or maybe adding a version which takes a ``zend_string`` might be another idea?

But to preserve API BC for extension, it might make sense to move this to a function named ``php_url_parse_ex3()`` and passing ``false`` as the strict parameter. This looks like it would prevent the diffs in most of the other extensions.

Reply:
fair enough I did that on purpose in case we would apply the strictness aspect in a couple of sub modules but I can backpedal on this, no problem.

Reply:
Maybe let's get some opinions from other people, @cmb69, @derickr ?

Reply:
oh ok I saw your comment too late